### ISSUE TITLE
Question: given a .zarr what is the best way to say that it is .ome.zarr / ngff?

### ISSUE THREAD

**Issue by yarikoptic** [ome/ngff#228]
is it presence of `.omero` key within `.zattrs`?

and then taking the schema of that version

```shell
❯ jq '.omero.version' .zattrs
"0.4"
```

correct?

**Comment by d-v-b** [comment]
Unfortunately, `.omero` is [optional](https://github.com/ome/ngff/blob/453570650eaec7f9d925a331f61bcf91aed950d4/0.4/index.bs#L441); The `multiscales` attribute might be better for this kind of thing. You might be interested in [this proposal](https://github.com/ome/ngff/pull/227) to fix the lack of a proper namespace in the metadata (among other changes).

**Comment by yarikoptic** [comment]
> You might be interested in [this proposal](https://github.com/ome/ngff/pull/227) to fix the lack of a proper namespace in the metadata (among other changes).

and there it would become "explicit" through use of a URL pointing to a versioned ngff?

```json
{
  "zarr_format": 3,
  "node_type": "group",
  "attributes": {
    "https://ngff.openmicroscopy.org/0.6": {
      "multiscales": [
```

or something else?

**Comment by joshmoore** [comment]
Yes, the single JSON file (zarr.json) would either have or not have the appropriate key.

**Comment by yarikoptic** [comment]
do you know if some more explicit formalization was discussed in zarr, e.g. to have  something like `"namespaces"` or `"extensions"` on the top level which would have contained smth like

```json
 [
  {
    "name": "ome",
    "version": "0.6",
    "schema": "https://ngff.openmicroscopy.org/jsonschemas/0.6" # 
  }
 ]
```

so, while dealing with a .zarr a developer of a tool could know what additional schemas implemented within that .zarr and potentially even load them to validate.  
Just making it some "unique versioned URL" requires knowledge of its correspondence to specific effort/project, and since URL doesn't point to schema (it is 404 ATM) - not immediately usable for validation etc.

**Comment by joshmoore** [comment]
It's being discussed to some degree in https://github.com/zarr-developers/zarr-specs/pull/262 ([ZEP0004](https://zarr.dev/zeps/draft/ZEP0004.html))

**Comment by yarikoptic** [comment]
any idea on how would it look for OME/NGFF if that ZEP0004 accepted in current form?

**Comment by thewtex** [comment]
We may want to encourage, i.e. (`SHOULD`) the use of `.ome.zarr` in the spec to improve awareness of OME-Zarr and be able to identify an OME-Zarr directly from a path / URL.

**Comment by thewtex** [comment]
> the use of .ome.zarr in the spec

Discussed by @normanrz at the 2024-04-03 OME Community Meeting: https://hackmd.io/So61knrQR0iLftd2LGAVjA

**Comment by yarikoptic** [comment]
> We may want to encourage, i.e. `SHOULD`, the use of `.ome.zarr` in the spec to improve awareness of OME-Zarr and be able to identify an OME-Zarr directly from a path / URL.

...

> > the use of .ome.zarr in the spec
> 
> Discussed by @normanrz at the 2024-04-03 OME Community Meeting: https://hackmd.io/So61knrQR0iLftd2LGAVjA

 @normanrz :

![image](https://github.com/ome/ngff/assets/39889/490f7bdb-9ad2-4374-b774-f070b0fb2fc5)

and I would wholeheartedly support that -- only for UX!
